packages/react-instantsearch/src/core/InstantSearch.js

```diff
@@ -150,13 +157,13 @@ class InstantSearch extends Component {
     } else {
       return '#';
     }
-  };
+  }
 
-  onHistoryInternalStateUpdate = state => {
+  onHistoryInternalStateUpdate(state) {
     this.aisManager.onExternalStateUpdate(state);
```

Comment:
How does that work? The `state =>` notation automatically binds to the current instance. In that case, since the `onHistoryInternalStateUpdate` method is passed by reference, it looses its context, so we need to have it bound to the `InstantSearch` instance somehow.